### PR TITLE
Show sparkle lightbulb on the end of a line containing only whitespace characters

### LINKED ISSUE #199458: Depending on the setting a different icon is shown
Testing #199158

For this code in an Untitled TypeScript editor:
```ts
let claims = undefined;
```

with setting set to `onCode`

Image

it's a lightbulb....

but when it's `on`:

Image

it's a sparkle.

behavior is the same

### PR DESCRIPTION
**Pull request by aiday-mar**
⚠️ merge after https://github.com/microsoft/vscode/pull/199786 is merged

Fixes https://github.com/microsoft/vscode/issues/199458

### CODE REVIEW

File: src/vs/editor/contrib/codeAction/browser/codeActionModel.ts [hunk 1]
```diff
@@ -72,7 +72,8 @@ class CodeActionOracle extends Disposable {
 		if (selection.isEmpty() && trigger.type === CodeActionTriggerType.Auto) {
 			const { lineNumber, column } = selection.getPosition();
 			const line = model.getLineContent(lineNumber);
-			if (line.length > 0) {
+			if (!/^\s*$/g.test(line)) {
```
Comment (aeschli) [inline src/vs/editor/contrib/codeAction/browser/codeActionModel.ts:75]: I think that check is ok. We just need to modify the behavior when we see that we are at the end of a line that is all whitespace

Comment (aiday-mar) [reply src/vs/editor/contrib/codeAction/browser/codeActionModel.ts:75]: So this check is related to the comment I posted here https://github.com/microsoft/vscode-copilot/pull/3165#discussion_r1411909632 and is the counterpart of the logic on the VS Code side.

File: src/vs/editor/contrib/codeAction/browser/codeActionModel.ts [hunk 2]
```diff
@@ -72,7 +72,10 @@ class CodeActionOracle extends Disposable {
 		if (selection.isEmpty() && trigger.type === CodeActionTriggerType.Auto) {
 			const { lineNumber, column } = selection.getPosition();
 			const line = model.getLineContent(lineNumber);
-			if (line.length === 0) {
+			if (
+				line.length === 0
+				|| (/^\s*$/g.test(line) && column - 1 === line.length)
```
Comment (aeschli) [inline src/vs/editor/contrib/codeAction/browser/codeActionModel.ts:77]: Isn't this duplicating the case a few lines below? 
https://github.com/microsoft/vscode/blob/5f9b8fa76331d69494e5adb75f8be48a9660cc6a/src/vs/editor/contrib/codeAction/browser/codeActionModel.ts#L89

Comment (aiday-mar) [reply src/vs/editor/contrib/codeAction/browser/codeActionModel.ts:77]: It's not the exact same check, but I'll join the checks

Comment (aiday-mar) [reply src/vs/editor/contrib/codeAction/browser/codeActionModel.ts:77]: It's not the exact same check, I will reuse what I can

**Pull request closed without merging**